Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,214 @@
+    } catch {
+        Write-Warning "Failed to collect drivers list: $_"
+    }
+}
+
+function Get-VirtioWinStorageDrivers {
+    $registryPaths = @(
+        'HKLM:\SYSTEM\CurrentControlSet\Services\Disk',
+        'HKLM:\SYSTEM\CurrentControlSet\Services\viostor\Parameters',
+        'HKLM:\SYSTEM\CurrentControlSet\Services\vioscsi\Parameters'
+    )
+    $valuesToQuery = @('IoTimeoutValue', 'TimeoutValue')
+
+    foreach ($path in $registryPaths) {
+        foreach ($value in $valuesToQuery) {
+            try {
+                $property = Get-ItemProperty -Path $path -Name $value -ErrorAction Stop
+                $output = "$path\$value : $($property.$value)"
+                $output | Out-File -FilePath 'virtio_disk.txt' -Append
+            } catch {
+                Write-Warning "Failed to collect ${path}\${value}: $_"
+            }
+        }
+    }
+}
+
+function Get-WindowsUpdateLogs {
+    try {
+        $logPath = Join-Path -Path (Get-Location) -ChildPath 'WindowsUpdate.log'
+        $command = "Get-WindowsUpdateLog -LogPath '$logPath'"
+        Start-Process -FilePath 'powershell.exe' -ArgumentList '-NoLogo', '-NoProfile', '-Command', $command -NoNewWindow -Wait -RedirectStandardOutput 'OutputWindowsUpdate.log' -RedirectStandardError 'ErrorWindowsUpdate.log'
+    } catch {
+        Write-Warning "Failed to collect Windows Update logs: $_"
+    }
+}
+
+function Get-WindowsUptime {
+    try {
+        $uptime = (Get-Date) - (gcim Win32_OperatingSystem).LastBootUpTime
+        $uptime.ToString() | Out-File -FilePath 'WindowsUptime.txt'
+        Write-Host 'Windows uptime collection completed.'
+    } catch {
+        Write-Warning "Failed to collect Windows uptime: $_"
+    }
+}
+
+function Get-ServicesList {
+    try {
+        Get-Service | Select-Object -Property Name, DisplayName, Status, StartType | Export-Csv -Path 'Services.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path 'RunningProcesses.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path 'InstalledApplications.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path 'InstalledKBs.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Export-WindowsMemoryDump {
```

Comment:
I suggested differentiating functions that generate files and return paths to existing files by prefixing `Export` for the former and prefixing `Get` for the latter.